rust_src/src/editfns.rs

```diff
@@ -152,3 +152,27 @@ pub fn goto_char(position: LispObject) -> LispObject {
     };
     position
 }
+
+/// Return the byte position for character position POSITION.
+/// If POSITION is out of range, the value is nil.
+#[lisp_fn]
+pub fn position_bytes(position: LispObject) -> LispObject {
+    if let LispNumber::Fixnum(pos) = position.as_number_coerce_marker_or_error() {
+        let cur_buf = ThreadState::current_buffer();
```

Comment:
It seems like we use `as_fixnum_coerce_marker_or_error` for `CHECK_NUMBER_COERCE_MARKER` and `as_number_coerce_marker_or_error` for `CHECK_NUMBER_OR_FLOAT_COERCE_MARKER`. I think this is a little irritating, what do you think @Wilfred.

Reply:
I totally missed `as_fixnum`. And yes, having it return a 'real' number and `as_number` returning a `LispNumber` is confusing.